### ISSUE TITLE
a load is completely broken - AUR0707:route,load

### ISSUE THREAD

**Issue by wzrdtales** [aurelia/aurelia#1835]
The following

```
<a load="route: routeid; params.bind: {id: modId}">
```

causes AUR0707:route,load, even removing params.bind, still causes this error. This seems to be completely broken. The route exists, it uses the normal router. Not sure nothing special here to add much.

Here is a very simple straightforward bootstrapped project, that reproduces this:

https://github.com/wzrdtales/au-repro-router-error
https://stackblitz.com/github/wzrdtales/au-repro-router-error

**Comment by wzrdtales** [comment]
also just 

```
<a load="route: routeid;">
```

results in the same

**Comment by wzrdtales** [comment]
in this example, it returns AUR0707: Bindable route not found on load. instead. also unexpected error which shouldn't happen at all

https://stackblitz.com/edit/au2-conventions-ab6pw8?file=src%2Fmy-app.html

**Comment by wzrdtales** [comment]
only working with `this.router.load` works, `a load="router:xyz;"` returns this error all the time.

i must say the experience is very very very frustrating

**Comment by wzrdtales** [comment]
according to https://docs.aurelia.io/routing/navigating#html-load-attribute this should work, it does not.

**Comment by wzrdtales** [comment]
https://github.com/wzrdtales/au-repro-router-error

**Comment by bigopon** [comment]
@jwx are you able to have a look at the stackblitz?

**Comment by jwx** [comment]
> @jwx are you able to have a look at the stackblitz?

Yes, will take a look as soon as possible. (Sorry for late reply, have been unavailable.)

**Comment by wzrdtales** [comment]
`id:` seems to work

**Comment by Sayan751** [comment]
@wzrdtales In case you haven't found a solution for the problem, and your app strictly uses configured routing (as opposed to direct routing), may I suggest that you give the router-lite a try? The docs are [here](https://docs.aurelia.io/getting-to-know-aurelia/routing/aurelia-router-lite/getting-started). If the router-lite also have the same problem, please let me know.

**Comment by wzrdtales** [comment]
declined. router-lite lacks features we use.

**Comment by Sayan751** [comment]
Noted. Just for my personal interest, may I ask what features those are?

**Comment by wzrdtales** [comment]
no solution, just a workaround.

we use this function everywhere and use click trigger with this combined instead of using the official functions...

```
 goto (route, parameters) {
   return this.router.load(route, { parameters })
}
```

**Comment by wzrdtales** [comment]
we make use of direct routing

**Comment by wzrdtales** [comment]
not sure if anyone is yet working on fixing this bug, but I guess someone is, as this is one of the very core functions that need to work properly.

**Comment by Sayan751** [comment]
> we make use of direct routing

Thank you for the clarification.

**Comment by migajek** [comment]
have the same problem. and I find it pretty serious too. 
it's even possible to reproduce it using the "recipe tutorial" example: 

1. open the stackblitz https://stackblitz.com/edit/au2-conventions-n56ej9
2. in my-app.ts append `id: 'home'` to the first route definition 
3. in my-app.html change the home link to `<a load="route: home">Home</a>` 
4. in the console, you'll get `AUR0707: Bindable route not found on load.`

**Comment by jwx** [comment]
Sorry for the long wait! Rough family events have prevented me from being as active as I'd like for quite some time now.  Gradually on my way back now.

Regarding the issue: In `router`, the `load` bindables are `id` and `parameters` (matching the properties in the configured route). @Vheissu , can you update the docs?

Please let me know if you run into any issues using these bindables.

**Comment by migajek** [comment]
It seems to work when I changed the "route" to "id".  Glad it's only docs issue. Thank you!
